### PR TITLE
[Network] Drop `azure-mgmt-network` SDK

### PR DESCRIPTION
**Pull request by kairu-ms**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by kairu-ms** [comment]
Test package size.

**Comment by kairu-ms** [comment]
@bebound Could you check why the MSI package size grows after I remove the `azure-mgmt-network`? Thanks.

**Comment by bebound** [comment]
`Trim_sdk.py` also need to be modified.
```
2023-02-14T05:59:30.2006655Z Traceback (most recent call last):
2023-02-14T05:59:30.2008027Z   File "D:\a\_work\1\s\scripts\trim_sdk.py", line 20, in <module>
2023-02-14T05:59:30.2008775Z     import azure.mgmt.network
2023-02-14T05:59:30.2009494Z ModuleNotFoundError: No module named 'azure.mgmt.network'
```

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py [hunk 1]
```diff
@@ -377,6 +377,7 @@ def _ensure_subnet_delegation(cli_ctx, subnet_id, delegation_service_name):
 
 
 def _ensure_route_table(cli_ctx, resource_group_name, ase_name, location, subnet_id, force):
+    from azure.mgmt.network.models import RouteTable, Route
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py:380]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py [hunk 2]
```diff
@@ -444,6 +444,7 @@ def _ensure_route_table(cli_ctx, resource_group_name, ase_name, location, subnet
 
 
 def _ensure_network_security_group(cli_ctx, resource_group_name, ase_name, location, subnet_id, force):
+    from azure.mgmt.network.models import NetworkSecurityGroup
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py:447]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py [hunk 3]
```diff
@@ -88,6 +88,7 @@ def test_app_service_environment_list(self, ase_client_factory_mock):
     @mock.patch('azure.cli.command_modules.appservice.appservice_environment._get_ase_client_factory', autospec=True)
     def test_app_service_environment_create(self, ase_client_factory_mock, subnet_show_mock,
                                             resource_client_factory_mock, deployment_name_mock):
+        from azure.mgmt.network.models import Subnet
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py:91]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py [hunk 4]
```diff
@@ -220,6 +220,7 @@ def test_app_service_environment_delete(self, ase_client_factory_mock, resource_
     @mock.patch('azure.cli.command_modules.appservice.appservice_environment._get_ase_client_factory', autospec=True)
     def test_app_service_environment_v3_create(self, ase_client_factory_mock, subnet_show_mock,
                                                resource_client_factory_mock, deployment_name_mock):
+        from azure.mgmt.network.models import (Subnet, Delegation)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py:223]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py [hunk 5]
```diff
@@ -262,6 +262,7 @@ def test_app_service_environment_v3_create(self, ase_client_factory_mock, subnet
     @mock.patch('azure.cli.command_modules.appservice.appservice_environment._get_ase_client_factory', autospec=True)
     def test_app_service_environment_v3_zone_create(self, ase_client_factory_mock, subnet_show_mock,
                                                resource_client_factory_mock, deployment_name_mock):
+        from azure.mgmt.network.models import (Subnet, Delegation)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_app_service_environment_commands_thru_mock.py:265]:
```suggestion
```

**Review by necusjz — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**